jwst/cube_build/cube_build_wcs_util.py

```diff
@@ -582,6 +648,82 @@ def set_geometryAB(self, footprint):
 
 
 
+#_______________________________________________________________________
+# from a set of ra values find the average ra.
+# This is tricky because of the convergence of hour angles
+# this method is taken from the SPITZER SSC MOSAICER tool
+def average_ra(ravalues):
+
```

Comment:
I was just taking a quick look through this, but did you consider using [astropy.units.Angle](http://docs.astropy.org/en/stable/coordinates/angles.html) and [astropy.stats.circmean](http://docs.astropy.org/en/stable/stats/circ.html)?

Reply:
I was trying to find something in astro py that did something like this. I thought it must exist but
I could not find it. I will look at those two routines in astropy.